pubsub/mqtt3/mqtt.go

```diff
@@ -45,8 +45,7 @@ type mqttPubSub struct {
 	topics          map[string]mqttPubSubSubscription
 	subscribingLock sync.RWMutex
 	reconnectCh     chan struct{}
-	ctx             context.Context
-	cancel          context.CancelFunc
+	closeCh         chan struct{}
```

Comment:
As discussed I like having a channel since we can select over it. Have added an atomic to prevent multiple closes.

Have also added a wait group to block `close()`.